homeassistant/components/smarla/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for Swing2Sleep Smarla integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from pysmarlaapi import Connection
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+from .const import DOMAIN, HOST
+
+STEP_USER_DATA_SCHEMA = vol.Schema({CONF_ACCESS_TOKEN: str})
+
+
+class SmarlaConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Swing2Sleep Smarla."""
+
+    VERSION = 1
+
+    async def _handle_token(self, token: str) -> tuple[dict[str, str], dict[str, str]]:
+        """Handle the token input."""
+        errors: dict[str, str] = {}
+        info: dict[str, str] = {}
+
+        try:
+            conn = Connection(url=HOST, token_b64=token)
+        except ValueError:
+            errors["base"] = "invalid_token"
+            return (errors, info)
+
+        if await conn.refresh_token():
+            info["serial_number"] = conn.token.serialNumber
+            info["token"] = conn.token.get_string()
```

Comment:
What's the difference between the token we get from here and the token itself we input?